// The proto specification for lockfile locks.
syntax = "proto2";

message LockFile {
  optional int32 owner_process_id = 1;
  optional string owner_process_argv = 2;
  optional int64 date_acquired_utc_epoch_ms = 3;

  // The hostname of the owning process. Stale locks cannot be detected if the
  // lock is owned by a different hostname, since we can't determine if the
  // owning process is alive or not.
  optional string owner_hostname = 4;
  // The user of the process which owns the lock. This has no effect on lock
  // acquisition, it is purely for documentation.
  optional string owner_user = 5;
}
